src/dparse/lexer.d

```diff
@@ -1526,6 +1526,23 @@ private pure nothrow @safe:
             index);
     }
 
+    bool lexNamedEntity()
+    {
+        assert (range.bytes[range.index] == '&');
+
+        Token t;
+        range.popFront();
+        lexIdentifier(t);
+        if (t.type != tok!"identifier" ||
+            !range.empty && range.bytes[range.index] != ';')
```

Comment:
D-Scanner would complain that you need to add () around the && exp here to clarify precedence.

You should also either write out the `!range.empty` as the comparision it does or change the `range.bytes[...]` call to use the range interface. (`.front`) Mixing random access and range API calls is prone to errors for modifications